### DISCUSSION TITLE
Nested scrolling

### DISCUSSION THREAD

**Discussion by hkparker** [fyne-io/fyne#5303] [General]
I have a vertical scroll that contains inside of it, among many other things, another vertical scroll.

When I'm scrolling down, as soon as my mouse enters the inner vertical scroll, all scrolling stops.  This is because my inner scroll actually doesn't have enough stuff in it to need scrolling, and you can't really tell it's a scroll.  Effectively, it's already scrolled all the way down.  But, trying to scroll down further doesn't result in any scrolling, my mouse is over that inner scroll now and it's capturing all my scroll intent.

It seems to me that once this inner scroll hits the bottom, any further attempts to scroll down should be passed up to the scroll above.  This lines up with other behaviors I've seen in other apps.  This is probably a pretty difficult thing to do in the current API.  Currently I just have to move my mouse out of the inner scroll to keep going once I get there, which is fine, but counter intuitive.

Thoughts?

**Comment by andydotxyz** [comment]
Yes the events will go to the widget under the cursor as you surmised.

Scroll within a scroll is not great UX (Apple documents this in their Human Interface Guidelines) and it fairly famously doesn't work on iOS in Safari (IIRC).

If the expectation is that it will work we should try and make that happen, but I don't know how...

**Reply by hkparker** [reply]
Yeah I definitely agree its not a great UX, just what I'm working with right now.  I'll probably do something different eventually, but either way, thought it was worth sharing this behavior in case there was a solution I was missing, and I was curious to see if this behavior was a goal.  I could understand it being a non-goal, though I do feel like I've seen it elsewhere.
